addons/mail/static/tests/mock_server/mail_mock_server.js

```diff
@@ -1077,105 +1086,387 @@ const ids_by_model = {
     Store: [],
 };
 
-const MANY = Symbol("MANY");
-const ONE = Symbol("ONE");
+export class StoreAttr {
+    constructor(name, value, predicate) {
+        const kwargs = unmakeKwArgs(getKwArgs(arguments, "name", "value", "predicate"));
+        this.name = kwargs.name;
+        delete kwargs.name;
+        this.value = kwargs.value;
+        delete kwargs.value;
+        this.predicate = kwargs.predicate;
+        delete kwargs.predicate;
+    }
+
+    _get_value(record, model = null) {
+        if (typeof this.value === "function") {
+            return this.value(record);
+        }
+        const value = this.value ?? record[this.name];
+        return value;
+    }
+}
+
+export class StoreRelation extends StoreAttr {
+    constructor(name_or_record, fields, value, predicate, as_thread, only_id) {
+        const kwargs = unmakeKwArgs(
+            getKwArgs(
+                arguments,
+                "name_or_record",
+                "fields",
+                "value",
+                "predicate",
+                "as_thread",
+                "only_id"
+            )
+        );
+        name_or_record = kwargs.name_or_record;
+        delete kwargs.name_or_record;
+        fields = kwargs.fields;
+        delete kwargs.fields;
+        value = kwargs.value;
+        delete kwargs.value;
+        predicate = kwargs.predicate;
+        delete kwargs.predicate;
+        as_thread = kwargs.as_thread;
+        delete kwargs.as_thread;
+        only_id = kwargs.only_id ?? false;
+        delete kwargs.only_id;
```

Comment:
As we have this pattern a lot, it might be worth having a function to extract + delete a list of fields from kwargs into a new kwargs object with only those keys.